### PR TITLE
Http compression

### PR DESCRIPTION
**Pull request by SherlySun**
Add support for http-compression

### CODE REVIEW

File: src/connection.cc [hunk 1]
```diff
@@ -133,9 +139,41 @@ Connection::ProcessRequest(const Request& request)
...
+      std::cout<<"Compressing using deflate"<<std::endl;
+      com.CompressDeflate();
+      response.SetBody(com.GetCompressedBody());
+
+    } else if(find(encode_format.begin(),encode_format.end(),"gzip") != encode_format.end()){
+      response.AddHeader("Content-Encoding","gzip");
+      std::cout<<"Compressing using gzip"<<std::endl;
+      com.CompressGzip();
+      response.SetBody(com.GetCompressedBody());
+    }
```
Comment (caraxin) [inline src/connection.cc]: you'd better change the header(Content-Length) after compression.

File: src/connection.cc [hunk 2]
```diff
@@ -133,9 +139,41 @@ Connection::ProcessRequest(const Request& request)
...
+
+    if(find(encode_format.begin(),encode_format.end(),"deflate") != encode_format.end()){
+      response.AddHeader("Content-Encoding","deflate");
+      std::cout<<"Compressing using deflate"<<std::endl;
+      com.CompressDeflate();
+      response.SetBody(com.GetCompressedBody());
+
+    } else if(find(encode_format.begin(),encode_format.end(),"gzip") != encode_format.end()){
+      response.AddHeader("Content-Encoding","gzip");
+      std::cout<<"Compressing using gzip"<<std::endl;
```
Comment (caraxin) [inline src/connection.cc]: Use BOOST_LOG_TRIVIAL(trace) <<

**Comment by SherlySun** [comment]
Just fix the content length problem. Nice suggestion there~

File: src/connection.cc (hunk 1 above)
Comment (SherlySun) [reply src/connection.cc]: Just fix the content length problem. Nice suggestion there~

File: src/connection.cc (hunk 2 above)
Comment (SherlySun) [reply src/connection.cc]: Oh, forgot to change this.

**Pull request merged**